### DISCUSSION TITLE
[Feature request] Money transfer log - Ingame Accessibility

### DISCUSSION THREAD

**Discussion by TheoRetisch1** [Zrips/CMI#1517] [Feature Requests]
The money transfer log is an important feature to detect abuse. Since not everyone in a server team has console access, the money log should be accessible through an ingame command. So I suggest the following changes:

1) There should be an ingame command /money log [player name] [page number] that lists the corresponding player-related entries in the chat.
2) There should be two permissions: cmi.command.moneylog and cmi.command.moneylog.others.
3) For better performance the entries should be stored in the mysql database and not in a txt file. The txt file becomes very fast, very long and is anything but efficient.
4) The money log should be integrated into the purge function, so that you can delete entries older than x-days automatically.

A money log would not only help the server team, but could also give normal players a cool overview of their financial history #statement of bank account!

**Comment by Zrips** [comment]
At this point money logs are separated into months, so you can always remove files which you dont need. For storage type, in general yml file is fine as we are not rewriting entire file but simply adding values into it. Its quite efficient way and simpler to handle then adding extra save batch for mysql and having database expanding even more. 
having command to check logs is a good idea tho.

**Comment by TheoRetisch1** [comment]
Thank you for the explanation and for considering! I look forward to having a command at some point 👍

**Comment by TheoRetisch1** [comment]
I'd like to bring this feature back to attention. An ingame available money log would still be a very cool and probably a frequently used feature! There is also another request (#2049).

The problem with data storage is not really solved either. For me the money log doesn't split up month by month. I have only one file, which is currently 29MB large. This amount of data is hard to handle, both for an future ingame money log, and for the private readout.

We read the file with a self-developed tool to give our supporters access to it. If not differentiated to a certain player it is only possible to display the data for a maximum of 8 days. The file ist just to large and the problem is getting bigger and bigger.

I would be happy about an update :)

**Comment by Zrips** [comment]
Damn, got confused with selllog, that one is separated by months. Need to change moneyLog to behave in same way.
